### ISSUE TITLE
[FEATURE] Request a secret

### ISSUE THREAD

**Issue by homotechsual** [HemmeligOrg/Hemmelig.app#182]
### Description

Add the ability to send a link requesting a secret that then gets shared back to a pre-configured destination (email/webhook)

**Comment by homotechsual** [comment]
We need to collect a secret from someone and want to do so securely. Hemmelig solves nicely for us sending ephemeral credentials to people but we don't want to allow unrestricted secret creation and we don't want to create accounts for every third party or end user we need a secret from so what we're looking for is:

We'd essentially be sending a Hemmelig link to someone which can bypass login requirements and allow them to create a secret with the link to the completed secret going back to the person who generated the initial link.

An example commercial tool serving this need: https://contentsnare.com/ 

This isn't a need we have often but enough that having a tool which can do this would be useful. Also open to frame challenge here - maybe we can do this already and just need to tweak how we plan to use Hemmelig.

**Comment by bjarneo** [comment]
I see. Seems like a great feature. Will come back to you on this. Thanks :)

**Comment by ltguillaume** [comment]
Yeah this would be fantastic! I'm currently using a Cryptpad form for such use cases, which is okay, but since Cryptpad doesn't support any sort of notifications and Hemmelig now _does_, it would be a fantastic feature to set up.

**Comment by bjarneo** [comment]
I am doing some testing here, but I might release a beta site to test this before it is going live. I'll add it to this issue if I do.

**Comment by bjarneo** [comment]
Please give it a spin on Hemmelig.app, will not release it to others until something is nailed: https://hemmelig.app/dashboard/secret-requests

**Comment by ltguillaume** [comment]
I like the way the requests are created, but maybe it would be nice to have the one sending the secret to me _only_ have me send the decryption key, not a complete URL?

Otherwise it doesn't really have any additional benefit compared to directing them to hemmelig.app.

It would guarantee that only I can view the secret via my dashboard and the user can send the decryption key via any communication channel, trusted or untrusted, because it doesn't include the secret's ID.

This would also reintroduce a feature lost in v7 to some extent (namely the split copying of URL and decryption key, in order to easily send them over separate channels), while keeping it as easy as possible for the other party.

**Comment by bjarneo** [comment]
I hear you, and I agree. I have been thinking about this myself. 

If a password is being used, then the decryptionKey is not needed. The password will be the decryption key.

I need to rethink the flow.

> This would also reintroduce a feature lost in v7 to some extent (namely the split copying of URL and decryption key, in order to easily send them over separate channels), while keeping it as easy as possible for the other party.

In v7, there are no encryption keys in combination with a password. The password is the decryption key.

**Comment by ltguillaume** [comment]
>In v7, there are no encryption keys in combination with a password. The password is the decryption key.

Honestly, I think that's a pretty bad idea.

1. People will choose a **bad password** almost every time
1. I _wanted_ to be able to use a bad password to keep things simple, and to function as **2FA**, i.e. sending that via another channel than the link
1. There doesn't seem to be any **brute force protection** for passwords. The IDs for the secrets are guessable to some extent and once you have one with a password, you can try it forever.

I really think the optional password only has value as additional measure, next to a decent decryption key.

**Comment by bjarneo** [comment]
For now, when the requested secret is created, it will display an input field with the decryption key for the requester to copy.

I might iterate on this feature later on.
